lib/i18n/middleware.rb

```diff
@@ -0,0 +1,15 @@
+module I18n
+  class Middleware
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      @app.call(env)
+    ensure
+      Thread.current[:i18n_config] = I18n::Config.new
```

Comment:
Doing this will cause the translations to be reloaded upon every single request. Since your issue seems to be just that the default locale is not used again for `I18n.locale`, why not have here instead:

```ruby
I18n.locale = I18n.default_locale
```

?

Reply:
@radar thanks for the review, I don't think it will cause the translations to be re-loaded, as `Backend::Simple` is initialized as a class variable and once rails is initialized, the same object is persisted/re-used:

https://github.com/svenfuchs/i18n/blob/6ab51c78be833eb011d5b4d3f5b2c54e74847512/lib/i18n/config.rb#L19

You can test this by going to `rails console` and checking that `backend`'s `object_id` is always the same:

```ruby
2.3.3 :003 > I18n::Config.new.backend.object_id
 => 70171618106860 
2.3.3 :004 > I18n::Config.new.backend.object_id
 => 70171618106860 
```

That said, since the main issue is of TLS in a multi-threaded environemt, updating `Thread.current` itself made sense :).